dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.42/V2_42_32__User_name_as_generated_column.sql

```diff
@@ -0,0 +1,11 @@
+/*
+Adds the "name" column to users as a computed column based on first and last name
+to move named based filters into the DB.
+This is important as the web API "query" search is a combination of multiple columns
+one of which is name. Such combinations must stay all in DB to be correct.
+
+Since both source columns are not null the name can and should also be not null
+and the computation does not need to handle null.
+*/
+alter table userinfo
+    add column if not exists name character varying(321) not null generated always as ( firstname || ' ' || userinfo.surname ) stored;
```

Comment:
any reason the table name prefix is used for `surname` and not `firstname` when generating the value? it looks a little funny.

Reply:
That must have been auto-complete insert I didn't notice. I'll remove the prefix.